[Safe CPP] Address Warnings in RenderSlider

stwrt opened the pull request:
#### 142dbe3af82c9832da0b962b31d8dfda395f01a2
<pre>
[Safe CPP] Address Warnings in RenderSlider
<a href="https://bugs.webkit.org/show_bug.cgi?id=293928">https://bugs.webkit.org/show_bug.cgi?id=293928</a>
<a href="https://rdar.apple.com/problem/152461354">rdar://problem/152461354</a>

Reviewed by Tim Nguyen.

Fix Safe CPP warnings in RenderSlider.

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/rendering/RenderSlider.cpp:
(WebCore::RenderSlider::element const):
(WebCore::RenderSlider::inDragMode const):
(WebCore::RenderSlider::valueRatio const):
* Source/WebCore/rendering/RenderSlider.h:

Canonical link: <a href="https://commits.webkit.org/295822@main">https://commits.webkit.org/295822@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/621d3fc9c7341c00a8e6ed7ae919cfb6bad6d32b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106237 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25986 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16381 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111435 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56833 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108276 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26648 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34489 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80697 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109241 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21074 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95865 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61024 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20609 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13966 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56273 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90430 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14001 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114295 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33375 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24609 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89770 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33739 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92096 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89471 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22819 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34339 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12151 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28955 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33300 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38712 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33046 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36396 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34644 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/accessibility/AccessibilitySlider.cpp:
```diff
@@ -171,7 +175,9 @@ LayoutRect AccessibilitySliderThumb::elementRect() const
         return LayoutRect();
-    if (auto* thumbRenderer = sliderRenderer->element().sliderThumbElement()->renderer())
+    CheckedRef element = sliderRenderer->element();
+    CheckedPtr sliderThumbElement = element->sliderThumbElement();
```
Why use the Checked variants? Elements can be ref-counted

nt1m replied:
Also this should use the `protected` or `checked` member functions rather than an extra local variable since there's only one usage of each in this method.

nt1m commented on Source/WebCore/rendering/RenderSlider.cpp:
```diff
@@ -105,16 +105,17 @@ void RenderSlider::computePreferredLogicalWidths()
 {
-    return element().sliderThumbElement()->active();
+    CheckedPtr sliderThumbElement = element()->sliderThumbElement();
+    return sliderThumbElement->active();
```
Same question here. You could just use `protectedSliderThumbElement()`

nt1m commented on Source/WebCore/rendering/RenderSlider.cpp:
```diff
@@ -60,7 +60,7 @@ RenderSlider::RenderSlider(HTMLInputElement& element, RenderStyle&& style)
 RenderSlider::~RenderSlider() = default;
 
-HTMLInputElement& RenderSlider::element() const
+CheckedRef<HTMLInputElement> RenderSlider::element() const
```
Usually we create a separate version called `checkedElement()` rather than changing the normal one.

(Also any reason this is not return a `Ref`? Elements can be ref-counted)

nt1m commented on Source/WebCore/rendering/RenderSlider.cpp:
```diff
@@ -105,16 +106,17 @@ void RenderSlider::computePreferredLogicalWidths()
 {
-    return element().sliderThumbElement()->active();
+    CheckedPtr sliderThumbElement = element().sliderThumbElement();
+    return sliderThumbElement->active();
```
Could this use `RefPtr` ? if so you could use this:
```suggestion
    return element.protectedSliderThumbElement()->active();
```

Otherwise, you should add a `checkedSliderThumbElement()` variant and call that, though using `RefPtr` would be preferable here.

nt1m commented on Source/WebCore/rendering/RenderSlider.cpp:
```diff
@@ -105,16 +107,18 @@ void RenderSlider::computePreferredLogicalWidths()
-    return element().sliderThumbElement()->active();
+    Ref element = this->element();
+    RefPtr sliderThumbElement = element->sliderThumbElement();
+    return sliderThumbElement->active();
```
According to style guide, this should either be:
`return element().protectedSliderThumbElement()->active();` or `return protectedElement()->protectedSliderThumbElement()->active();`. 

First one if `protectedSliderThumbElement()` is a trivial getter, second one if not.

nt1m commented on Source/WebCore/rendering/RenderSlider.cpp:
```diff
@@ -105,16 +107,16 @@ void RenderSlider::computePreferredLogicalWidths()
 bool RenderSlider::inDragMode() const
 {
-    return element().sliderThumbElement()->active();
+    return this->element().protectedSliderThumbElement()->active();
```
I don't think you need `this->`

nt1m commented on Source/WebCore/rendering/RenderSlider.cpp:
```diff
@@ -105,16 +107,17 @@ void RenderSlider::computePreferredLogicalWidths()
 {
-    return element().sliderThumbElement()->active();
+    Ref element = this->element();
+    return element->protectedSliderThumbElement()->active();
```
The guideline is to add a new `protectedElement()` method to `RenderSlider` returning a Ref, and then use that. We don't use local variables if there's only one usage.

stwrt committed:
[Safe CPP] Address Warnings in RenderSlider
https://bugs.webkit.org/show_bug.cgi?id=293928
rdar://problem/152461354

Reviewed by Tim Nguyen.

Fix Safe CPP warnings in RenderSlider.

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/rendering/RenderSlider.cpp:
(WebCore::RenderSlider::element const):
(WebCore::RenderSlider::inDragMode const):
(WebCore::RenderSlider::valueRatio const):
* Source/WebCore/rendering/RenderSlider.h:

Canonical link: https://commits.webkit.org/295822@main
